scripts/Muon/GUI/Common/pairing_table_widget/pairing_table_widget_presenter.py

```diff
@@ -126,9 +126,23 @@ def add_pair_to_view(self, pair):
         self._view.enable_updates()
 
     def handle_add_pair_button_clicked(self):
-        pair = self._model.construct_empty_pair(self._view.num_rows() + 1)
-        self.add_pair(pair)
-        self.notify_data_changed()
+        new_pair_name = self._view.enter_pair_name()
+        if new_pair_name in self._model.group_and_pair_names:
+            self._view.warning_popup("Groups and pairs must have unique names")
+        elif self.validate_pair_name(new_pair_name):
+            if len(self._model.group_names) == 1:
```

Comment:
I don't think we want to allow pairs to be created which use the same group twice as when we attempt to calculate these pairs will fail. Can we instead just display the warning popup in the case there is only one group saying something along the lines of you need two groups to make a pair.